src/utils.ts

```diff
@@ -0,0 +1,135 @@
+import { Dayjs, OpUnitType } from "dayjs";
+import _ from "lodash";
+
+export const deepCopy = <T>(value: T): T => {
+  return _.cloneDeep(value);
+};
+
+export const isContained = <T extends string | number | boolean>(
+  value: unknown,
+  array: T[]
+): value is T => {
+  return array.findIndex((el) => el === value) !== -1;
+};
+
+export const initializeObject = <K extends string | number | symbol, V>(
+  keys: K[],
+  initValue: V
+): Record<K, V> => {
+  return keys.reduce(
+    (obj, key) => ({ ...obj, [key]: deepCopy(initValue) }),
+    {} as Record<K, V>
+  );
+};
+
+export const getKeysFromObj = <T extends object>(obj: T) =>
+  Object.keys(obj) as (keyof T)[];
+
+export const removeDuplicate = <T>(array: T[]): T[] =>
+  new Array(...new Set(array));
+
+export const isEqualPrimitive = <T extends string | number | boolean>(
+  v1: T,
+  v2: T
+): boolean => v1 === v2;
+
+export const isEqualSet = <T>(s1: Set<T>, s2: Set<T>): boolean => {
+  return s1.size === s2.size && [...s1].every((el) => s2.has(el));
+};
+
+export const createId = (): string => {
+  return (
+    new Date().getTime().toString(16) +
+    "-" +
+    Math.floor(Math.random() * 100000)
+      .toString(16)
+      .padStart(5, "0")
+  );
+};
+
+export const isBetween = (
+  target: Dayjs,
+  start: Dayjs,
+  end: Dayjs,
+  unit?: OpUnitType
+): boolean => {
+  return (
+    (start.isSame(target, unit) || start.isBefore(target, unit)) &&
+    (end.isSame(target, unit) || end.isAfter(target, unit))
+  );
+};
+
+/**
+ * Whether the element is contained in the array.
+ */
+export const containedInArray = <T extends { id: string }>(
+  array: T[],
+  element: T
+): boolean => {
+  return array.findIndex(({ id }) => id === element.id) !== -1;
+};
+
+export const addElementsInArray = <T extends { id: string }>(
```

Comment:
これはこの PR じゃなくてもいいのですが、忘れないうちに
```suggestion
export const addElementsToArray = <T extends { id: string }>(
```